presto-main/src/main/java/com/facebook/presto/execution/QueryStateMachine.java

```diff
@@ -575,13 +581,27 @@ public boolean transitionToFinishing()
 
     private boolean transitionToFinished()
     {
+        try {
+            metadata.cleanupQuery(session);
```

Comment:
We had a disagreement about how does it should be, please have a look: https://github.com/prestodb/presto/pull/6114#issuecomment-260452150